topics/ui/controls/checkbox.topic

```diff
@@ -0,0 +1,194 @@
+    <tldr>
+        <p>
+            <control>Implementation:</control>
+            <a href="https://docs.oracle.com/javase/tutorial/uiswing/components/button.html"><code>JCheckBox</code></a>,
+            <a href="%gh-ic%/platform/platform-api/src/com/intellij/ui/components/JBCheckBox.java"><code>JBCheckBox</code></a>
+        </p>
+    </tldr>
+    <img src="checkbox.png" alt="A preview of checkboxes in different states" width="706"/>
+    <chapter title="When to use" id="when-to-use">
+        <p>Use checkboxes for yes/no choices or for selecting several items in a group:</p>
+        <img src="checkbox_when_to_use.png" alt="An example of a checkbox group with two selected options" width="706"/>
+    </chapter>
+    <chapter title="When not to use" id="when-not-to-use">
+        <p>Use <a href="radio_button.md">radio button</a> instead if:</p>
+        <table style="none" border="false">
+            <tr>
+                <td width="50%">
+                    <img src="checkbox_when_not_to_use_1.png" alt="A radio button group with only one selected option possible" width="378"/>
+                </td>
+                <td>Only one option in a group can be selected</td>
+            </tr>
+            <tr>
+                <td width="50%">
+                    <img src="checkbox_when_not_to_use_2.png" alt="Two radio buttons with clear labels" width="378"/>
+                </td>
+                <td>The behavior in the "off" state is unclear from the checkbox label. Use two radio buttons instead
+                    and label them accordingly
+                </td>
+            </tr>
+            <tr>
+                <td width="50%">
+                    <img src="checkbox_when_not_to_use_3.png" alt="A radio button group displaying the correct way for showing different states of a setting" width="378"/>
+                </td>
+                <td>With the checkbox, it is unclear how the setting works if it’s unchecked. With radio buttons, both
+                    states are labeled clearly
+                </td>
+            </tr>
+        </table>
+    </chapter>
+    <chapter title="How to use" id="how-to-use">
+        <chapter title="Label" id="label">
+            <table style="none" border="false">
+                <tr>
+                    <td width="50%"><img src="checkbox_when_to_use_1.png"
+                                         alt="A checkbox showing the correct way to use a label" width="378"/></td>
+                    <td>A label accompanies each checkbox and is placed next to it.</td>
+                </tr>
+                <tr>
+                    <td width="50%"><img src="checkbox_when_to_use_2.png"
+                                         alt="A checkbox with a long label split into two lines" width="378"/></td>
+                    <td>If a label is long, split it into two lines. Use HTML formatting for that.</td>
+                </tr>
+            </table>
+            <tabs group="languages">
```

Comment:
Maybe wrap the tabs into a separate chapter named "Split labels", and collapse it? https://www.jetbrains.com/help/writerside/collapsible-elements.html#collapsible_chapter

Reply:
Good idea, I've made it as an Implementation collapsible chapter